src/BenchmarkDotNet/ConsoleArguments/MisspellingsFinder.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.ConsoleArguments
+{
+    public class MisspellingsFinder
+    {
+        private readonly List<string> benchmarkNames = new List<string>();
+
+        public MisspellingsFinder(IReadOnlyList<Type> benchmarks)
+        {
+            foreach (var benchmark in benchmarks)
+            {
+                benchmarkNames.AddRange(benchmark.GetMethods()
+                    .Where(methodInfo => methodInfo.HasAttribute<BenchmarkAttribute>())
+                    .Select(methodInfo => $"{benchmark.FullName}.{methodInfo.Name}"));
+            }
+
+            benchmarkNames.AddRange(benchmarks.Select(bn => bn.FullName).Distinct());
+            benchmarkNames.AddRange(benchmarks.Select(bn => bn.Namespace).Distinct());
+            benchmarkNames.AddRange(benchmarks.Select(bn => bn.Name).Distinct());
+        }
+
+        public string[] Find([NotNull] string userInput)
```

Comment:
Since the updated implementation actually sorts benchmarks and move most relevant results to the top, I'm not sure that `MisspellingsFinder.Find` is the best name for this API. Could we come up with better name?

Reply:
@AndreyAkinshin is CorrectionsSuggester ok?